### ISSUE TITLE
_initWorked not making false on stop() & cancel() methods

### ISSUE THREAD

**Issue by vikshith7** [csdcorp/speech_to_text#338]
Once a user wants to stop/cancel 
![image](https://user-images.githubusercontent.com/59953021/187693568-b8a39370-8696-430f-bc91-c4c3075f8361.png)
checking through _initWorked once users stop need to make _initWorked = false on both cancel & stop methods. Please check on this issue. if it is not please close it with a particular comment/solution.

**Comment by sowens-csd** [comment]
Sorry, I don't understand what problem you're trying to solve with this suggested code change. Can you explain the behaviour you're seeing that you think is incorrect?

**Comment by vikshith7** [comment]
When I made a call for this method stop() or cancel() it is calling one private method _stop() in the first condition checking for **if(!_initWorked)**
if speech is already initialised, _initWorked is true, so I need to stop/cancel the initial condition will fail because 
if(!_initWorked) this condition returns false and the method channel will call to stop the speech that works as expected but _initWorked value needs to make false because the value is true while checking. please debug it done the above scenario.
this need to add here.
![image](https://user-images.githubusercontent.com/59953021/187842335-5f9e202a-9f9f-43c8-975a-958f20344917.png)

**Comment by sowens-csd** [comment]
The addition you have suggested is not correct. The check for initialization `_initWorked` is to see if the `initialize` method has been called and succeeded before the `stop` behaviour is triggered. But calling `stop` doesn't change the initialization state of the plugin, it stops the current `listen` session. The flow is something like this:

`initialize -> listen -> stop -> listen -> cancel -> listen -> stop...`

In other words you can call `listen` and `stop` or `cancel` many times after each `initialize` call but you should call `initialize` only once for the lifetime of the plugin. It is safe to call `stop` or `cancel` even if there is no active listening session. There is no way to uninitialize the plugin. 

Does that answer your question?

**Comment by vikshith7** [comment]
Yep, I got the point & answer for my question too.
Am expecting this PR https://github.com/csdcorp/speech_to_text/pull/133 to merge because this is also affecting some scenarios when can it be merged?

**Comment by sowens-csd** [comment]
I'm going to close that PR because I think you can already achieve those goals just by reassigning the error or status listener after initialization. That would look this:

```dart
      speech.errorListener =
          (errorNotification) => print(errorNotification.errorMsg);
      speech.statusListener = (statusNotification) =>
          print('statusListener: ${statusNotification}');
```

Let me know if this doesn't handle the cases you're interested in.
